### PR TITLE
HCL: hide sensitive variables from output

### LINKED ISSUE #9939: HCL2: sensitive strings are not removed from output:
```hcl
variable "sensitive" {
  default = "I am soooo sensitive"
}

source "null" "first-example" {
  communicator = "none"
}

build {
  name = "roles"

  source "null.first-example" {
    name = "consul"
  }

  provisioner "shell-local" {
    inline = ["echo ${source.name} and ${source.type}, the sensitive var: ${var.sensitive}"]
  }
}
```

```shellsession
$ packer build file.pkr.hcl
roles.null.consul: output will be in this color.

==> roles.null.consul: Running local shell script: /var/folders/3k/2gb5ct4s7cncr52_jh2kz6cw0000gq/T/packer-shell881229952
    roles.null.consul: consul and null, the sensitive var: I am soooo sensitive
...
```

I think we need to make sure they are strings and then pass them to the core as sensitive vars.

### PR DESCRIPTION
**Pull request by azr**
This allows to obfuscate sensitive variables from output by outputting `"<sensitive>"` in place.

fixes #9939

### CODE REVIEW

**Review by nywilken — APPROVED** [review]
Nicely done. Looks good to me

```
⇶  packer.test build null-sensitive-variables.json
null: output will be in this color.

==> null: Running local shell script: /tmp/packer-shell235025836
    null: the consul and null, sensitive var: <sensitive>
Build 'null' finished after 4 milliseconds 515 microseconds.

==> Wait completed after 4 milliseconds 544 microseconds

==> Builds finished. The artifacts of successful builds are:
--> null: Did not export anything. This is the null builder

⇶  packer.test build null-sensitive-variables.pkr.hcl
roles.null.consul: output will be in this color.

==> roles.null.consul: Running local shell script: /tmp/packer-shell415740413
    roles.null.consul: consul and null, the sensitive var: <sensitive>
Build 'roles.null.consul' finished after 7 milliseconds 531 microseconds.

==> Wait completed after 7 milliseconds 575 microseconds

==> Builds finished. The artifacts of successful builds are:
--> roles.null.consul: Did not export anything. This is the null builder

```

**Comment by nywilken** [comment]
@azr I approved this PR already but I realized that I wanted to request that you update the documentation for the new sensitive variable option. By either adding an HCL2 version to https://www.packer.io/docs/templates/user-variables#sensitive-variables or by adding the documentation to https://www.packer.io/docs/from-1.5/variables

File: command/test-fixtures/hcl-inspect-with-sensitive-vars/vars.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+variable "sensitive_array" {
+  default   = ["Im supersensitive", "me too !!!!"]
+  sensitive = true
+}
+
+variable "sensitive_tags" {
+  default   = {
+      first_key  = "this-is-mega-sensitive"
```
Comment (SwampDragons) [inline command/test-fixtures/hcl-inspect-with-sensitive-vars/vars.pkr.hcl:21]: love these vars

**Review by SwampDragons — COMMENTED** [review]
Just needs docs! This is great and thanks for pairing on it.

**Comment by azr** [comment]
Good catch on the missing docs, there it is, and it looks like this: Screenshot 2020-10-05 at 15 22 33

File: command/test-fixtures/hcl-inspect-with-sensitive-vars/vars.pkr.hcl (hunk 1 above)
Comment (azr) [reply command/test-fixtures/hcl-inspect-with-sensitive-vars/vars.pkr.hcl:21]: all the 5 shades of 'sensitive' ✅

**Pull request merged**